pkg/storer/internal/transaction/mem.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2024 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package transaction
+
+import (
+	"github.com/ethersphere/bee/v2/pkg/storage"
+	"github.com/ethersphere/bee/v2/pkg/storage/storageutil"
+)
+
+type memcache struct {
+	storage.IndexStore
+	data map[string][]byte
```

Comment:
Be careful with using a map to add a large number of keys as it will internally grow the number of map buckets and even when the keys are deleted the number of buckets will not shrink. This can lead to a "data leak" manifested as more memory used by the map even if it is cleared.

Reply:
the file is not needed at this point, removed